File: home_assignments/java/TicketSeller/TicketSellerServices/src/main/java/ticketseller/services/interfaces/IFestivalService.java
package ticketseller.services.interfaces;

import ticketseller.model.entities.Festival;
import ticketseller.model.entities.Ticket;
import ticketseller.services.ServiceException;

import java.time.LocalDate;
import java.util.Collection;

public interface IFestivalService {

    Collection<Festival> getAllFestivals();

    Collection<Festival> getAllFestivalsOnDate(LocalDate date) throws ServiceException;

    Ticket sellTicket(Integer festivalID, String buyerName, Integer spots) throws ServiceException;
}

File: home_assignments/csharp/TicketSeller/services/interfaces/IFestivalService.cs
using System;
using System.Collections.Generic;
using model.entities;

namespace services.interfaces;

public interface IFestivalService
{
    ICollection<Festival> GetAllFestivals();
    
    ICollection<Festival> GetAllFestivalsOnDate(DateTime date);
    
    Ticket SellTicket(int festivalId, String buyerName, int spots);

}